### DISCUSSION TITLE
[Public Testing] rAudio - Stable Candidates

### DISCUSSION THREAD

**Discussion by rern** [rern/rAudio#1371] [Testing]
[Edit 20231024] Pending and yield for normal releases.

- This release includes all updates
- Any issues regarding these images should be reported within this thread.

**Comment by kw-sudo** [comment]
Hi @rern ,

thank you for providing.

Installed clean from the scratch, no pacman -Syu update. Just did all the settings, added NAS and a complete library rebuild (only _'album sorted by artist'_).

The result is still unchanged. All 'singles' (tagged as such, thus no album tag) are displayed in album view, regardless which view settings. In i20230521 only the albums are shown.

![library](https://github.com/rern/rAudio/assets/104136223/c08c7cc8-73a2-4cca-8c72-679887990e42)

Oliver

**Reply by kw-sudo** [reply]
Based on the screenshot above, when I click on an album to add it, the subsequent view still only shows the cover art. The titles are 'hidden' behind the image and I have to scroll down to get an overview.

Oliver

**Comment by Rasb74** [comment]
Hi Rern, 

List of what I see not working :

1 - Power button
Working but rain blinking even after a full boot.

2 - Json parse when using « back screen » feature on iPhone. Working on iPad and computer.

3 - « Latest » albums, only the latest 15 uploaded are shown, is it the original setup or a limitation. 

Cheers
Pascal

**Comment by ceejayemm** [comment]
@rern 
I thought I had posted this earlier, so apologies if this ends up as a duplicate.

I installed the new 'stable' version on a RPi3A and generally has worked fine, however:

- all the issues reported earlier in #1356 are still an issue in this version
- in Features / Others / Browser on RPi is always initially enabled by default, even when a display is not attached to the RPi
- the Library option to enable 'Album - Sort by Artist' doesn't remain enabled consistently (despite the work of #1347) so the sort order of the library flip/flops between sort orders.

I also suspect, but can't prove at the moment, that some settings are being stored in the Shared Data area (which I use) and thus are being propagated to other systems using the same Shared Data area and are then being incorrectly interpreted by the other systems. I think some further work needs to be done to ensure that settings for one system, other than the shared Library, don't end up being inadvertently used by other systems.

I would also like to say a very BIG thank you to Rern for all his hard work on this project.  He has put a lot of work into some massive refactoring work over the last few months and I can see why/how some of the current issues might have come to light. I know some users don't like this kind of thing being done to 'production' systems hence the recent request for a 'stable' version. Myself, I am happy to work in both areas by running both a Main/Update and stable version on different RPis.

I can see why the idea of a 'stable' release seems useful to some people and I will be interested to see how this pans out. I suspect that this will end up being the release of a specific image which has had the bugs knocked out of it in the Update and possibly the Main release areas.  It also be interested to see how those users who who regularly apply the 'pacman -Syu' updates to their installations fare with this 'stable' release as I can't see how Rern will be able to cope with the possible newer dependencies of this kind of update being introduced when he does not know about them.

Chris

**Reply by kw-sudo** [reply]
The (re)installation of r/Audio is no big deal for me and to support rern's work I like to test new updates promptly. If necessary, even more often ;-)

@ceejayemm
I forgot to add this information. With the pre-release image a pacman -Syu update is not necessary, respectively there is "nothing to do".

Basically, compared to a pure r/audio installation, I could not see any big differences with a supplementary pacman - update.  

Consciously however only once:
After installing the 20230521 image, the library rebuild ran into a "json error". If I did a pacman -Syy sync before, it ran error free.

So far
Oliver

**Comment by markblinkhorn** [comment]
I did a clean install with the Pre-release candidate on a R-Pi 2 (BCM2836) and still have the same issue with CIFS attached NAS music library as described in **rAudio 20230609 NAS behaviour change #1346**

Mark

**Comment by hjfisch** [comment]
Hello ,
I' ve a jason error when I play a Song with special Characters in the Name like  ' .
In this example the Song is written  Breezin'.flac .
rAudio plays the song and displays  the jason-error message .
The next song will be played normal without that message .
Sadly I don't know in which Version of rAudio this error appears first .
B.t.w. I did a clean install with this release .

```
Errors: JSON.parse: bad escaped character at line 91 column 82 of the JSON data 
{ 
  "player"       : "mpd"
, "btreceiver"   : false
, "card"         : 0
, "consuXme"      : false
, "control"      : ""
, "counts"       : {
  "album": 5198,
  "albumartist": 2122,
  "albumyear": 5338,
  "artist": 2121,
  "composer": 0,
  "conductor": 4,
  "dabradio": 0,
  "date": 252,
  "genre": 374,
  "latest": 0,
  "nas": 0,
  "playlists": 0,
  "sd": 0,
  "song": 59777,
  "usb": 1,
  "webradio": 23
}
, "icon"         : ""
, "librandom"    : false
, "lyrics"       : false
, "relays"       : false
, "relayson"     : false
, "scrobble"     : false
, "shareddata"   : false
, "snapclient"   : false
, "stoptimer"    : false
, "updateaddons" : false
, "updating_db"  : false
, "updatingdab"  : false
, "volume"       : 100
, "volumemute"   : 0
, "webradio"     : false
, "display"          : {
    "plclear": false,
    "plsimilar": false,
    "audiocdplclear": true,
    "album": true,
    "nas": false,
    "albumartist": true,
    "sd": false,
    "artist": false,
    "usb": true,
    "composer": false,
    "playlists": false,
    "conductor": false,
    "webradio": true,
    "date": false,
    "genre": false,
    "count": true,
    "latest": true,
    "label": true,
    "albumbyartist": true,
    "albumyear": false,
    "tapaddplay": false,
    "tapreplaceplay": true,
    "playbackswitch": true,
    "backonleft": false,
    "hidecover": false,
    "fixedcover": true,
    "bars": true,
    "barsalways": true,
    "time": true,
    "radioelapsed": true,
    "cover": true,
    "covervu": false,
    "volume": false,
    "vumeter": false,
    "buttons": true,
    "noswipe": false
, "audiocd"          : true
, "camilladsp"       : false
, "color"            : "128 100 50"
, "dabradio"         : false
, "equalizer"        : false
, "multiraudio"      : false
, "relays"           : false
, "screenoff"        : false
, "snapclient"       : false
, "volumenone"       : true
}
, "repeat" : false
, "random" : false
, "single" : false
, "file"      : "USB/Greeto/hjfSound/Blank & Jones/Relax Edition 01/01 - Breezin\`.flac"
, "pllength"  : 25
, "song"      : 0
, "state"     : "play"
, "timestamp" : 1687846135872
, "stream" : false
, "Album"  : "Relax Edition 01"
, "Artist" : "Blank & Jones"
, "Time"   : 207
, "Title"  : "Breezin\`"
, "ext"      : "FLAC"
, "coverart" : ""
, "icon"     : ""
, "sampling" : "1/25 • 16 bit 44.1 kHz 648 kbit/s • FLAC"
, "elapsed"  : 38
, "coverart" : "%2Fmnt%2FMPD%2FUSB%2FGreeto%2FhjfSound%2FBlank%20%26%20Jones%2FRelax%20Edition%2001%2FFolder.jpg"
, "login" : false }```

**Reply by rern** [reply]
Change filename and Title: Breezin\\\` to Breezin' (or Breezin` if prefer backtick)

**Comment by hjfisch** [comment]
The filename is Breezin ' .
There is no backslash in the name .
The backslash is only shown in the jason error .
No backslash either in the playlist and it worked with former versions of rAudio .

**Reply by kw-sudo** [reply]
rern did not mean 'backslash', but an accent (french accent "grave" --> `). You use a apostrophe. :-)

Oliver

**Reply by rern** [reply]
All right. The bug has been found. To be fixed by next update.

**Comment by hjfisch** [comment]
> All right. The bug has been found. To be fixed by next update.
Thank's , building a stable fork is a very good idea .
For me , rAudio has all the functions I need , so that a bugfree version would be nice .

**Comment by HJLK** [comment]
My configuration: 
- Raspberry Pi 3B Rev 1.2
- HifiBerry DAC+
- CharLCD via PCF8574
- RotaryEncoder for play/pause and volume up/down
- button for Boot (pin5)
- button for Shutdown (pin7)
- startup.sh and shutdown.sh in /boot (in which via Python an extra LED on pin37 is switched ON/OFF)

I used image **20230620** to start. I put a backup.gz from 20230616 in /boot, but this did NOT work. After I put my wifi-file in /boot all went OK. Of course I now had to adjust System Settings for my components, and after a reboot **all my components work as expected**! 

I Modified:
- Buffer - Audio = 8192
- Buffer - Output = 16384
- FFMPEG ON

- Browser on RPi OFF
- Equalizer ON
- AutoPlay ON

- Lyrics still OFF: this does NOT default work anymore, since 202305, and I don't know why... When Lyrics ON, some internet address is needed, and I have no idea. **Why is this changed**?

- Equalizer ON: does work in audio, but picture is **disformed** (10 bands but only 8 slots...)
- PROBLEM: MUTE-button does not function: volume-ball tracks to zero, but percentage stays the same, just as the volume itself. This happens in both gui-alternatives (non-wheel and wheel), in web gui and app.

So far, so good, a stable version is much appreciated! Thanks Rern!

**20230630**:
Did a pacman -Syu, and rebooted. All things still work ok.
Then I upgraded rAudio to 20230630, and rebooted. 
Now running:
**rAudio 20230630**
**6.1.35-4 aarch64**

FINDINGS (browser gui and app gui):
- Equalizer image now OK
- The MUTE button still does not function as expected.

===============================================
2023-08-25 there is little to no progress in main branche since 2023-06-30. I noticed rern was busy with branche _Camilladsp_. Yesterday I noticed rern has committed _Camilladsp_ to _main_, but I cannot update to it because the Update-button is not (yet?) active! Can anybody comment on this? 

2023-08-29 The UPDATE-button was usable today, so I updated to version 230828. Findings sofar (after reboot):

- the MUTE button functions now as expected, in web/app GUI.

- **stopping** via button and web/app GUI is instantaneous. But starting is not: the music starts, after a small second there is a pause in sound for about a second. This is not a new "bug", by the way. It did occur also in previous releases. 

- the "lyrics"-feature is not repaired, nor is it clear what to do to get lyrics back...

**Reply by Roland-B** [reply]
@HJLK 

May want to look at this for your Lyrics issue.
https://github.com/rern/rAudio/discussions/1277#discussioncomment-6106436

Roland.

**Reply by HJLK** [reply]
Thanks @Roland-B, I had seen this already, but no one has posted a solution yet. And no one, not even Rern, has made clear why this (default) option is abandoned...

**Reply by HJLK** [reply]
================================================
Just updated to rAudio 20230918 / 6.1.35-4 aarch64, did a pacman -Syu which ended in ERROR:

(189/189) checking for file conflicts              [######################] 100%
error: failed to commit transaction (conflicting files)
camilladsp: /etc/default/camilladsp exists in filesystem
camilladsp: /usr/lib/systemd/system/camilladsp.service exists in filesystem (owned by camillagui-backend)
Errors occurred, no packages were upgraded.


Findings rAudio sofar are still the same:

-  stopping via button and web/app GUI is instantaneous. But starting is not: the music starts, after a small second there is a pause in sound for about a second. This is not a new "bug", by the way. It did occur also in previous releases. NB: this does not occur when playing mp3's, only with radio stations.

- the "lyrics"-feature is not repaired, nor is it clear what to do to get lyrics back...
 
=================================================
20230922 I corrected things for pacman -Syu, and it worked as far as I can see, also after reboot. I'm now running:
**rAudio 20230918
6.1.54-1 aarch64**

Findings for rAudio are still the same however...

**20231005** Just updated to version 20231001, rebooted, and did a pacman -Syu, Now on:
**rAudio 20231001
6.1.54-4 aarch64
Raspberry Pi 3B Rev 1.2
BCM2837 • 906 MiB
4 x Cortex-A53 @ 1.20 GHz**

Findings for rAudio are still the same, Though the pause in sound when starting radio is shorter. 
And still no one can tell me what happened with lyrics... I have 'm switched on, but what should I fill in the parameters box?

20231014 Just update to version 202310131, and I wish I hadn't!

- LCD display and web gui are not showing what is played with radio: lcd en web gui show the radio display, not the music anymore. With my mp3's it still does. So what's up, doc ?

**20231015**: and now (after  a reboot? Strange, because I rebooted yesterday after the update) it is working all normally again.
- except the Lyrics of course, that does not work anymore
-

**Comment by markblinkhorn** [comment]
Just for the record; I found placing backup.gz in /boot didn't work either. However, I restored the exact same file from the Sytem menu after the initial boot and everything (library and settings) came back as expected.

Mark Blinkhorn

________________________________

**Reply by HJLK** [reply]
@markblinkhorn: what do you mean with "the exact same file from the Sytem menu"? And where did you restore it?

**Reply by markblinkhorn** [reply]
@HJLK  - backup.gz was created with the Backup Option of the System Menu and downloaded to my PC. I copied that file to /boot (where it was ignored) and then selected **the same file** with the Restore Option of the System Menu. AKA - the exact same file.

Hope this helps. Mark

![image](https://github.com/rern/rAudio/assets/11545819/68b068ca-1569-46c1-8445-3141094b2caa)

**Reply by HJLK** [reply]
Thanks @markblinkhorn, I understand now what you meant. So I tried this myself, and still it did not work, probably because my backup.gz was from an earlier version...

**Comment by hjfisch** [comment]
Hello ,
My problem is solved in the new update , Thank's

**Comment by Roland-B** [comment]
Thanks for attempting to create a stable release image. 

**Issue 1** - Still unable to use E30 USB DAC with Spotify, shows as playing (spotify album cover shows) but the audio fails - no output.

<details>
<summary>Click to expand</summary>

Jun 27 21:18:59 rAudio systemd[1]: Started A spotify playing daemon.
Jun 27 21:18:59 rAudio spotifyd[3393]: Loading config from "/etc/spotifyd.conf"
Jun 27 21:18:59 rAudio spotifyd[3393]: No username specified. Checking username_cmd
Jun 27 21:18:59 rAudio spotifyd[3393]: No username_cmd specified
Jun 27 21:18:59 rAudio spotifyd[3393]: No password specified. Checking password_cmd
Jun 27 21:18:59 rAudio spotifyd[3393]: No password_cmd specified
Jun 27 21:18:59 rAudio spotifyd[3393]: No proxy specified
Jun 27 21:18:59 rAudio spotifyd[3393]: Using alsa volume controller.
Jun 27 21:18:59 rAudio spotifyd[3393]: no usable credentials found, enabling discovery
Jun 27 21:18:59 rAudio spotifyd[3393]: Failed to register IPv6 receiver: Os { code: 97, kind: Uncategorized, message: "Address family not supported by protocol" }
Jun 27 21:19:00 rAudio spotifyd[3393]: Ignoring blacklisted access point ap-gew4.spotify.com:4070
Jun 27 21:19:00 rAudio spotifyd[3393]: Ignoring blacklisted access point ap-gue1.spotify.com:443
Jun 27 21:19:00 rAudio spotifyd[3393]: Connecting to AP "ap-gew1.spotify.com:443"
Jun 27 21:19:01 rAudio spotifyd[3393]: Authenticated as "_Removed for privacy_" !
Jun 27 21:19:01 rAudio spotifyd[3393]: Country: "GB"
Jun 27 21:19:01 rAudio spotifyd[3393]: Converting with ditherer: tpdf
Jun 27 21:19:01 rAudio spotifyd[3393]: Using AlsaSink with format: S16
Jun 27 21:19:01 rAudio spotifyd[3393]: Running "/srv/http/bash/spotifyd.sh" using "/bin/bash" with environment variables {"PLAYER_EVENT": "volumeset", "VOLUME": "64386"}
Jun 27 21:19:01 rAudio spotifyd[3393]: Loading <Ruff Ryders' Anthem - Re-Recorded> with Spotify URI <spotify:track:51Llej1dJBMBY7z24YOLOr>
Jun 27 21:19:01 rAudio spotifyd[3393]: Running "/srv/http/bash/spotifyd.sh" using "/bin/bash" with environment variables {"PLAYER_EVENT": "start", "TRACK_ID": "51Llej1dJBMBY7z24YOLOr", "PL>
Jun 27 21:19:02 rAudio spotifyd[3393]: <Ruff Ryders' Anthem - Re-Recorded> (231040 ms) loaded
Jun 27 21:19:02 rAudio spotifyd[3393]: Audio Sink Error Invalid Parameters: <AlsaSink> Device hw:0 Unsupported Format S16LE (S16), ALSA function 'snd_pcm_hw_params_set_format' failed with >
Jun 27 21:19:02 rAudio systemd[1]: spotifyd.service: Main process exited, code=exited, status=1/FAILURE
Jun 27 21:19:02 rAudio systemd[1]: spotifyd.service: Failed with result 'exit-code'.

Changing settings to use device = "**default:CARD=E30**", allows it to work until the device is restarted and the settings file is re-written with wrong settings that do not work again :(

Todays update **rAudio 20230630** made no difference and still does not work except this time I have error in the update :

<img src="https://github.com/rern/rAudio/assets/43496391/140cc2a4-08d9-4d06-83a3-1f23667858d8)" width=300>

rAudio:~  #  aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: E30 [E30], device 0: USB Audio [USB Audio]
  Subdevices: 0/1
  Subdevice #0: subdevice #0
rAudio:~  #  aplay -L
null
    Discard all samples (playback) or generate zero samples (capture)
default
    Default Audio Device
sysdefault
    Default Audio Device
default:CARD=E30
    E30, USB Audio
    Default Audio Device
sysdefault:CARD=E30
    E30, USB Audio
    Default Audio Device
front:CARD=E30,DEV=0
    E30, USB Audio
    Front output / input
surround21:CARD=E30,DEV=0
    E30, USB Audio
    2.1 Surround output to Front and Subwoofer speakers
surround40:CARD=E30,DEV=0
    E30, USB Audio
    4.0 Surround output to Front and Rear speakers
surround41:CARD=E30,DEV=0
    E30, USB Audio
    4.1 Surround output to Front, Rear and Subwoofer speakers
surround50:CARD=E30,DEV=0
    E30, USB Audio
    5.0 Surround output to Front, Center and Rear speakers
surround51:CARD=E30,DEV=0
    E30, USB Audio
    5.1 Surround output to Front, Center, Rear and Subwoofer speakers
surround71:CARD=E30,DEV=0
    E30, USB Audio
    7.1 Surround output to Front, Center, Side, Rear and Woofer speakers
iec958:CARD=E30,DEV=0
    E30, USB Audio
    IEC958 (S/PDIF) Digital Audio Output

</details>



================================================

**Issue 2** **FIXED** - Webradio Add-on installed but still shows as available to install in add-ons.

================================================

**Issue 3** - Random folders with multiple albums in them do not give any option when selecting the folder such as Add, Play Next, Replace etc.

================================================

**Issue 4** - Unable to save auto downloaded album images, says doesn't have permission to write. The CIFS account/pw used to connect has full rights to everything, and worked previously a while back (cant say when it broke as don't use too often) with no changes being made to my share.

Regards,

Roland.
rAudio 20230630
6.1.35-3 aarch64
Raspberry Pi 4B Rev 1.2

**Reply by rern** [reply]
Hi Roland,

 Please try the UPDATE branch:
Issue 1  - A new option to select output for Spotify
Issue 2 - Fixed
Issue 3 - More detail?
Issue 4 - The target directory must be `777` / `drwxrwxrwx`

**Reply by Roland-B** [reply]
Hi Rern,

Thanks v much for the the hard work to try and remediate :)

**Issue 1** - I now have option to select the correct option thanks, but when then selecting the correct card/option the setting does not get commited (saved) and when check back it has reverted to "hw:0" again.

<details>
<summary>Click to expand</summary>

<img src="https://github.com/rern/rAudio/assets/43496391/89ca132d-48b8-4c79-a227-e92c85ce8c72)" width=300>

journalctl output below : 

Jul 03 16:36:20 rAudio sudo[6837]: http : PWD=/srv/http ; USER=root ; COMMAND=/srv/http/bash/settings/features-data.sh
Jul 03 16:36:20 rAudio sudo[6837]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=33)
Jul 03 16:36:20 rAudio systemd[1]: Binding to IPv6 address not available since kernel does not support IPv6.
Jul 03 16:36:20 rAudio systemd[1]: Binding to IPv6 address not available since kernel does not support IPv6.
Jul 03 16:36:20 rAudio systemd[1]: Binding to IPv6 address not available since kernel does not support IPv6.
Jul 03 16:36:20 rAudio systemd[1]: Binding to IPv6 address not available since kernel does not support IPv6.
Jul 03 16:36:20 rAudio sudo[6837]: pam_unix(sudo:session): session closed for user root
Jul 03 16:36:24 rAudio sudo[6916]: http : PWD=/srv/http ; USER=root ; COMMAND=/srv/http/bash/settings/features-data.sh
Jul 03 16:36:24 rAudio sudo[6916]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=33)
Jul 03 16:36:24 rAudio systemd[1]: Binding to IPv6 address not available since kernel does not support IPv6.
Jul 03 16:36:24 rAudio systemd[1]: Binding to IPv6 address not available since kernel does not support IPv6.
Jul 03 16:36:24 rAudio systemd[1]: Binding to IPv6 address not available since kernel does not support IPv6.
Jul 03 16:36:24 rAudio systemd[1]: Binding to IPv6 address not available since kernel does not support IPv6.
Jul 03 16:36:24 rAudio sudo[6916]: pam_unix(sudo:session): session closed for user root
Jul 03 16:36:25 rAudio sudo[6995]: http : PWD=/srv/http ; USER=root ; COMMAND=/srv/http/bash/settings/features.sh spotifyoutput
Jul 03 16:36:25 rAudio sudo[6995]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=33)
Jul 03 16:36:25 rAudio sudo[6995]: pam_unix(sudo:session): session closed for user root
Jul 03 16:36:28 rAudio sudo[7015]: http : PWD=/srv/http ; USER=root ; COMMAND=/srv/http/bash/settings/features.sh 'spotifyoutputset#012default:CARD=E30#012CMD OUTPUT'
Jul 03 16:36:28 rAudio sudo[7015]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=33)
Jul 03 16:36:28 rAudio systemd[1]: Stopping MPD idle...
Jul 03 16:36:28 rAudio systemd[1]: mpdidle.service: Deactivated successfully.
Jul 03 16:36:28 rAudio systemd[1]: Stopped MPD idle.
Jul 03 16:36:28 rAudio systemd[1]: Stopping Music Player Daemon...
Jul 03 16:36:28 rAudio systemd[1]: mpd.service: Deactivated successfully.
Jul 03 16:36:28 rAudio systemd[1]: Stopped Music Player Daemon.
Jul 03 16:36:28 rAudio systemd[1]: Starting Music Player Daemon...
Jul 03 16:36:29 rAudio mpd[7070]: decoder: Decoder plugin 'wildmidi' is unavailable: configuration file does not exist: /etc/timidity/timidity.cfg
Jul 03 16:36:29 rAudio systemd[1]: Started Music Player Daemon.
Jul 03 16:36:29 rAudio systemd[1]: Started MPD idle.
Jul 03 16:36:29 rAudio systemd[1]: Stopping A spotify playing daemon...
Jul 03 16:36:29 rAudio systemd[1]: spotifyd.service: Deactivated successfully.
Jul 03 16:36:29 rAudio systemd[1]: Stopped A spotify playing daemon.
Jul 03 16:36:29 rAudio systemd[1]: Started A spotify playing daemon.
Jul 03 16:36:29 rAudio spotifyd[7090]: Loading config from "/etc/spotifyd.conf"
Jul 03 16:36:29 rAudio spotifyd[7090]: No username specified. Checking username_cmd
Jul 03 16:36:29 rAudio spotifyd[7090]: No username_cmd specified
Jul 03 16:36:29 rAudio spotifyd[7090]: No password specified. Checking password_cmd
Jul 03 16:36:29 rAudio spotifyd[7090]: No password_cmd specified
Jul 03 16:36:29 rAudio spotifyd[7090]: No proxy specified
Jul 03 16:36:29 rAudio spotifyd[7090]: Using alsa volume controller.
Jul 03 16:36:29 rAudio spotifyd[7090]: no usable credentials found, enabling discovery
Jul 03 16:36:29 rAudio spotifyd[7090]: Failed to register IPv6 receiver: Os { code: 97, kind: Uncategorized, message: "Address family not supported by protocol" }
Jul 03 16:36:30 rAudio sudo[7015]: pam_unix(sudo:session): session closed for user root

</details>

**Issue 2** - Confirmed is fixed, thanks.

**Issue 3** - Worked out the issue. Not sure if it is expected behavior or not. If you have a folder which contains multiple albums (also in folders) and there are any music files [mp3, m4a etc] in the root of that folder then you cannot click anything and get an option to play, replace etc.

**Issue 4** - I have added "rw,file_mode=0777,dir_mode=0777" to the additional options for mounting which seems to force the permissions my NAS share already has for the user account/password I'm using to connect to my NAS through rAudio1. Is there any way you can revert to previous functionality where it is was not necessary to have to do this. I don't have to alter in Volumio or other Pi builds? Thanks.

Regards,

Roland.
rAudio 20230630 [Update] branch
6.1.35-3 aarch64
Raspberry Pi 4B Rev 1.2

**Reply by markblinkhorn** [reply]
@Roland-B - I can confirm that your 'issue 3' is the same issue I have been having - reported at #1346. I was starting to believe that it was something unique to my setup. However, It was not an issue prior to rAudio 20230609 and I have not been able to workaround it as yet.

Mark

**Reply by markblinkhorn** [reply]
@Roland-B 
I done a hack for the Context Menu problem - see #1346 for more info

**Reply by Roland-B** [reply]
@markblinkhorn. Bit late viewing this but thanks for that appreciated :)

**Reply by Roland-B** [reply]
@rern Just to add Spotify is now broken when starting from scratch on latest builds..  get "Invalid Grant" when trying to register the key.

**Comment by kw-sudo** [comment]
With 20230630 update, the mainscreen submenu (bio/lyrics/add…) is cut off at iOS/do not resize. I have to turn screen sideway to close. „x-button“

![IMG_5316](https://github.com/rern/rAudio/assets/104136223/4cb00432-be26-4c0e-9bc8-c4a7f9597e40)

Oliver

**Reply by rern** [reply]
Hi Oliver,

Please try the UPDATE branch.

**Reply by kw-sudo** [reply]
Hi rern,

I did so, still same issue (iOS 16.5.1)

[edit]: only with iPhone, iPad 10.2“ display is large enough to compensate (if necessary?)

**Comment by marcelgsteiger** [comment]
Thanks @rern for all your valuable work.
I'm always having problems getting Wifi Station to work from the originally flashed system or when the WiFi network changes. This remains the same with this new version. Putting the SSID/Password into BOOT/wifi as instructed does not work, putting backup.gz there does not work either. I believe this might be a WiFi Country or rfkill / "airplane mode" problem (country = Switzerland/CH). On a RPI Zero 2W I have to connect a USB-LAN Interface first, then configure the Wifi Country from the LAN, then manually do a modprobe brcmfmac using ssh - then the WiFi interface comes hopefully up. On Rpi 4B the system also starts without WiFi interface by default - no possibility to configure SSID/PW when accessing the system from the LAN interface.
When I can get Wifi to work once, everything seems stable across reboots then, though.
Perhaps we should have a possibility to configure the wifi country settings in the BOOT/wifi configuration file, or something similar.
Best regards
Marcel

**Reply by ceejayemm** [reply]
@marcelgsteiger 

My go to way of configuring wifi for rAudio (and in fact other applications which require wifi ona RPi) across a number of different RPi models (Zero, 3A, 3b, 3b+ and 4B) is to place a file called 'wifi' (with no extension) in the BOOT partition and works (for me) every time. My wifi file contains:


Interface=wlan0
Connection=wireless
IP=dhcp
ESSID="{ssid name}"                    <--- Replace {ssid name} with your own network name, note the quotes
Key={your password}                   <---Replace {your password} with your network password, note NO quotes
Security=wpa

Note: currently, restoring a backup file requires that the backup file was produced by a rAudio installation which is at the same version level as the system being restored to.

Chris

**Reply by marcelgsteiger** [reply]
@ceejayemm Thanks a lot, chris! I took the sample wifi0 file that contains quotes both for SSID and Key. Omitting the quotes for the key did the trick.
